src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexWriter.cs

```diff
@@ -160,14 +166,16 @@ private void CreateIndexWriter(IState state)
             using (_indexWriter.MergeScheduler)
             {
             }
-            _indexWriter.SetMergeScheduler(new SerialMergeScheduler(), state);
+
+            _mergeScheduler = new CustomSerialMergeScheduler();
+            _indexWriter.SetMergeScheduler(_mergeScheduler, state);
```

Comment:
Can we require to pass `_mergeScheduler ` to the constructor of `new CustomIndexWriter` instead of setting it this way?

Reply:
done